learn/src/main/java/org/navgurukul/learn/ui/learn/CourseDetailActivity.kt

```diff
@@ -0,0 +1,112 @@
+            intent.putExtra(ARG_KEY_COURSE_ID, courseId)
+            intent.putExtra(ARG_KEY_COURSE_NAME, courseName)
+            context.startActivity(intent)
+        }
+
+        var masterData: List<Exercise> = mutableListOf()
+    }
+
+    private lateinit var courseId: String
+    private lateinit var courseName: String
+    private lateinit var mBinding: ActivityCourseDetailBinding
+    private lateinit var mAdapter: CourseExerciseAdapter
+    private val viewModel: LearnViewModel by viewModel()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_course_detail)
+        parseIntentData()
+        initToolBar()
+        initExpandableToolBar()
+        initRecyclerView()
+        fetchData()
+    }
+
+
+    private fun parseIntentData() {
+        if (intent.hasExtra(ARG_KEY_COURSE_ID) && intent.hasExtra(ARG_KEY_COURSE_NAME)) {
+            courseId = intent.getStringExtra(ARG_KEY_COURSE_ID)!!
+            courseName = intent.getStringExtra(ARG_KEY_COURSE_NAME)!!
+        }
+    }
+
+    private fun initToolBar() {
+        setSupportActionBar(mBinding.toolbar)
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+        supportActionBar?.setHomeButtonEnabled(true)
+    }
+
+    private fun initExpandableToolBar() {
+        mBinding.toolbarLayout.setExpandedTitleColor(toolbarColor())
+        mBinding.toolbarLayout.setCollapsedTitleTextColor(toolbarColor())
+        mBinding.toolbarLayout.title = courseName
+    }
+
+
+    private fun initRecyclerView() {
+        mAdapter = CourseExerciseAdapter {
+            if (!it.first.slug.isNullOrBlank())
+                CourseSlugDetailActivity.start(
+                    this,
+                    CurrentStudy(
+                        courseId, courseName, it.first.slug!!, it.first.name, it.first.id
+                    )
+                )
+        }
+        val layoutManager =
+            LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false)
+        mBinding.contentCourseDetail.recyclerviewCourseDetail.layoutManager = layoutManager
+        mBinding.contentCourseDetail.recyclerviewCourseDetail.adapter = mAdapter
+    }
+
+
+    private fun fetchData() {
+        viewModel.fetchCourseExerciseData(courseId).observe(this, Observer {
+            mBinding.contentCourseDetail.progressBar.visibility = View.VISIBLE
+            if (null != it && it.isNotEmpty()) {
+                mBinding.contentCourseDetail.progressBar.visibility = View.GONE
+                masterData = it
+                mAdapter.submitList(it)
+            }
+        })
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        if (item.itemId == android.R.id.home)
+            finish()
+        return super.onOptionsItemSelected(item)
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        masterData = mutableListOf()
```

Comment:
assigning an empty list on onDestroy ? We should clear this list instead right ?

Reply:
```
 override fun onDestroy() {
        super.onDestroy()
        masterData.clear()
}
```